docs/baremetal-agent-controller.md

```diff
@@ -0,0 +1,155 @@
+Baremetal Agent Controller (a.k.a BMAC)
+==
+
+BMAC is a Kubernetes controller responsible for reconciling [BareMetalHost][bmo] and Agent (defined
+and maintained in this repo) resources for the agent-based deployment scenario.
+
+Testing
+==
+
+The testing environment for BMAC consists in:
+
+- [Downstream dev-scripts](https://github.com/openshift-metal3/dev-scripts/) deployment
+- [Baremetal Operator][bmo]: It defines the BareMetalHost custom resource
+- [Assisted Installer Operator](./operator.md): To deploy and manage the  assisted installer
+  deployment. Read the operator docs to know more about its dependencies and installation process.
+
+
+Each of the components listed above provide their own documentation on how to deploy and configure
+them. However, you can find below a set of recommended configs that can be used for each of these
+components:
+
+Dev Scripts
+===
+
+```bash
+# One worker, with enough memory and CPU, should be enough.
+# Use more if resources are not a problem
+export NUM_WORKERS=1
+# Add extra workers so we can use it for the deployment
+export NUM_EXTRA_WORKERS=1
+# At the time of this writing, this requires the 1195 PR
+# mentioned below.
+export PROVISIONING_NETWORK_PROFILE=Disabled
+# By default dev-scripts uses the metal3 BMO image
+# clone the downstream version and set the path here.
+export BAREMETAL_OPERATOR_LOCAL_IMAGE=/path/to/your/local/clone
+```
+
+The config above should provide you with an environment that is ready to be used for the operator,
+assited installer, and BMAC tests. Here are a few tips that would help simplifying the environment
+and the steps required:
+
+- Apply [1195](https://github.com/openshift-metal3/dev-scripts/pull/1195) if you haven't already
+- Clone [baremetal-operator][bmo] somewhere and set the variable in your
```

Comment:
Incomplete sentence. Clone somewhere and set BAREMETAL_OPERATOR_LOCAL_IMAGE to the directory